apps/src/templates/sectionProgress/sectionProgressLoader.js

```diff
@@ -142,63 +109,29 @@ function processStudentTimestamps(timestamps) {
   return studentTimestamps;
 }
 
-function processStudentPairing(students) {
-  const studentPairing = getStudentPairing(students);
-  const isValid = Object.keys(studentPairing).every(userId =>
-    Object.keys(studentPairing[userId]).every(
-      levelId => typeof studentPairing[userId][levelId] === 'boolean'
-    )
-  );
-  if (!isValid) {
-    throw new Error('Input is invalid');
-  }
-  return studentPairing;
-}
-
 function postProcessDataByScript(scriptData) {
-  if (!scriptData.stages) {
-    return scriptData;
+  // Filter to match scriptDataPropType
+  const filteredScriptData = {
```

Comment:
Could leverage `_.pick` https://lodash.com/docs/4.17.15#pick

Reply:
i recently ran into a case where i was refactoring some code that leveraged `pick` and found it cumbersome, since it only handles "picking" but not "mapping", so as soon as i started needing anything other than a direct mapping for a property i had to break it out from the `pick` call and things started getting messier. in this case, we'd already need to break out `stages: scriptData.lessons`, and imo this just feels cleaner (and more maintainable) than starting with a `pick` call then handling custom mappings afterward.

if we lived in a world where server shapes === frontend shapes i think `pick` would be much more useful, but in our case i don't think it's a helpful pattern to introduce/encourage.